tests/test-defaults.js

```diff
@@ -213,6 +213,35 @@ tape('recursive defaults', function(t) {
   })
 })
 
+tape('recursive defaults requester', function(t) {
+  t.plan(4)
+
+  var defaultsOne = request.defaults({}, function(options, callback) {
+      var headers = options.headers || {}
+      headers.foo = 'bar1'
+      options.headers = headers
+
+      request(options, callback)
+    })
+    , defaultsTwo = defaultsOne.defaults({}, function(options, callback) {
+      var headers = options.headers || {}
+      headers.baz = 'bar2'
+      options.headers = headers
+
+      defaultsOne(options, callback)
+    })
+
+  defaultsOne.get(s.url + '/get_recursive1', function (e, r, b) {
+    t.equal(e, null)
+    t.equal(b, 'TESTING!')
```

Comment:
That is being asserted over here: https://github.com/tikotzky/request/blob/fix-recursive-requester/tests/test-defaults.js#L65